workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/DashboardPage/DashboardPage.tsx

```diff
@@ -39,38 +38,61 @@ export function DashboardPage() {
   		return monthNames[monthIndex] + ', ' + day + ' ' + year;
 	}
 
-	const getProductTypeList = async () => {
-		const products = await getProducts();
+	function getAppListProductSpecifications(productIds : number[]) {
+		let appListProductSpecifications : any[] = [];
 
-		const productTypeList : string[] = [];
+		productIds.forEach((productId) =>  {
+			appListProductSpecifications.push(getProductSpecifications({appProductId: productId}));
+		})
+
+		return Promise.all(appListProductSpecifications);
+	}
+
+	function getAppListProductIds(products: any) {
+		const productIds : any[] = [];
 
 		products.items.map((product: any) => {
-			const productId : number = product.productId;
+			productIds.push(product.productId);
+		})
+
+		return productIds;
+	}
+
+	function getProductTypeFromSpecifications(specifications: any) {
+		var productType = 'no type';
 
-			getProductSpecifications({appProductId : productId}).then(productSpecifications =>
-				productTypeList.push(productSpecifications.items[0].value.en_US)
-			);
+		specifications.items.forEach((specification: any) => {
+			if (specification.specificationKey === "type") {
+				productType = specification.value.en_US;
+			}
 		})
 
-		return productTypeList;
+		return productType;
 	}
 
 	useEffect(() => {
-		(async () => {
-			let [products, productTypeList] = await Promise.all([getProducts(), getProductTypeList()]);
-
-			const liferayApps = products.items.map((product: any, index: number) => ({
-				thumbnail: product.thumbnail,
-				name: product.name.en_US,
-				updatedDate: formatDate(product.modifiedDate),
-				updatedBy: product.catalogId,
-				version: product.version,
-				status: product.productStatus,
-				type: productTypeList[index]
-			}))
-			setApps(liferayApps);
-		})();
-	}, []);
+		const setNewAppList = async () => {
```

Comment:
Is there a reason why we removed the IIFE?  Do we need to access this function elsewhere?